src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -55,10 +55,10 @@ internal sealed partial class ReflectTypeDescriptionProvider : TypeDescriptionPr
         // in.  The keys to the property and event caches are types.
         // The keys to the attribute cache are either MemberInfos or types.
         //
-        private static volatile Hashtable s_propertyCache;
-        private static volatile Hashtable s_eventCache;
-        private static volatile Hashtable s_attributeCache;
-        private static volatile Hashtable s_extendedPropertyCache;
+        private static readonly Lazy<Hashtable> s_propertyCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
+        private static readonly Lazy<Hashtable> s_eventCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
+        private static readonly Lazy<Hashtable> s_attributeCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
+        private static readonly Lazy<Hashtable> s_extendedPropertyCacheLazy = new Lazy<Hashtable>(() => new Hashtable());
```

Comment:
I think all of these would be better off using LazyInitializer.EnsureInitialized rather than Lazy.